src/main/java/org/wikipedia/login/LoginClient.java

```diff
@@ -61,7 +61,7 @@ public void onFailure(@NonNull Call<MwQueryResponse> call, @NonNull Throwable ca
         });
     }
 
-    void login(@NonNull final WikiSite wiki, @NonNull final String userName, @NonNull final String password,
+    public void login(@NonNull final WikiSite wiki, @NonNull final String userName, @NonNull final String password,
```

Comment:
If you would like to start using `LoginClient` now, you can use the `loginBlocking()` method below (and put it in an Rx `Action`). This `login()` method is not intended for external use.

Reply:
I tried [using the login client's](https://github.com/commons-app/apps-android-commons/pull/2894/files#diff-27435a827f3c50a6c24b839ba9322424R254) `loginBlocking()` method but realized that the `WIKIPEDIA_URL` is used inside the method. 

Am thinking of writing our own `LoginClient` in Commons app code. Let me know if you think we should somehow reuse the library's client.

Reply:
@dbrant I tried using the `loginBlocking()`  method and login works well. But how do I consume the login response to get the `username`. The `loginBlocking()` method doesn't return anything.